packages/compiler/src/lib/decorators.ts

```diff
@@ -343,6 +344,83 @@ export function isErrorModel(program: Program, target: Type): boolean {
   return false;
 }
 
+// -- @mediaTypeHint decorator --------------
+
+const [_getMediaTypeHint, setMediaTypeHint] = useStateMap<MediaTypeHintable, string>(
+  createStateSymbol("mediaTypeHint"),
+);
+
+/**
+ * A type that can have a default MIME type.
+ */
+type MediaTypeHintable = Parameters<MediaTypeHintDecorator>[1];
+
+export const $mediaTypeHint: MediaTypeHintDecorator = (
+  context: DecoratorContext,
+  target: MediaTypeHintable,
+  mediaType: string,
+) => {
+  validateDecoratorUniqueOnNode(context, target, $mediaTypeHint);
+
+  setMediaTypeHint(context.program, target, mediaType);
+};
+
+/**
+ * Get the default media type hint for the given target type.
+ *
+ * This value is a hint _ONLY_. Emitters are not required to use it, but may use it to get the default media type
+ * associated with a TypeSpec type.
+ *
+ * @param program - the Program containing the target
+ * @param target - the target to get the MIME type for
+ * @returns the default media type hint for the target, if any
+ */
+export function getMediaTypeHint(program: Program, target: Type): string | undefined {
+  switch (target.kind) {
+    case "Scalar":
+      // This special-casing is necessary because we cannot apply a decorator with a `valueof string` argument to
```

Comment:
if we are special casing this one could we maybe special case bytes too or you think augment is better?

as much as it work I do not want to recommend people augment the intrinsic types. This could be an execption but it does set precedent. What do you think?

Reply:
I think it's reasonable that people will want to augment intrinsic scalars with metadata. For example a library might want to define some metadata about types and in its lib.tsp apply that metadata through decorators to the intrinsic scalars so that it can be accessed through a uniform interface. We could special case bytes and avoid the augmentation if we don't want it in our core, but I don't think we should take a hard stance against augmenting the intrinsic scalars for this reason.

Reply:
yeah it make sense if you had a similar decorator and the lbirary took care of that but bad things can be done from a spec point of view